src/Jackalope/Version/VersionHistory.php

```diff
@@ -214,9 +231,52 @@ public function hasVersionLabel($label, $version = null)
      */
     public function getVersionLabels($version = null)
     {
-        // @codeCoverageIgnoreStart
-        throw new NotImplementedException();
-        // @codeCoverageIgnoreEnd
+        $this->initVersionLabels();
+        if ($version === null) {
+            return array_keys($this->versionLabels);
+        } else {
+            $versions = $this->getAllVersions();
+            $versionIsInHistory = false;
+
+            foreach ($versions as $versionCheck) {
+                /* @var VersionInterface $versionCheck */
+                if ($versionCheck->getIdentifier() == $version->getIdentifier()) {
+                    $versionIsInHistory = true;
```

Comment:
once we found that the version is in the list, we could `break;` to stop the foreach loop.